pkg/utils/utils_linux_test.go

```diff
@@ -91,3 +95,29 @@ func TestTcpID(t *testing.T) {
 	AddTcpID(f, uint64(1234))
 	assert.EqualValues(t, GetTcpID(f), uint64(1234))
 }
+
+func TestGetOutgoingInterface_RouteResolves(t *testing.T) {
+	const (
+		InterfaceName  = "Eth0"
+		InterfaceIndex = 42
+	)
+
+	knownLink := &netlink.Veth{LinkAttrs: netlink.LinkAttrs{Name: InterfaceName, Index: InterfaceIndex}}
+
+	routeList = func(_ netlink.Link, _ int) ([]netlink.Route, error) {
+		return []netlink.Route{
+			{LinkIndex: 1, Dst: &net.IPNet{IP: net.IPv4(192, 168, 0, 0), Mask: net.IPv4Mask(0xff, 0xff, 0xff, 0)}},
```

Comment:
For 'Mask' net.IPMask{255, 255, 255, 0} would be more readable and consistent with the decimal format used for 'Dst'